master/pkg/searcher/workload.go

```diff
@@ -20,12 +22,18 @@ const (
 
 // Workload encompasses a single unit of work that a trial needs do before waiting for more work.
 type Workload struct {
-	Kind         Kind `json:"kind"`
-	ExperimentID int  `json:"experiment_id"`
-	TrialID      int  `json:"trial_id"`
-	StepID       int  `json:"step_id"`
+	Kind             Kind `json:"kind"`
+	ExperimentID     int  `json:"experiment_id"`
+	TrialID          int  `json:"trial_id"`
+	StepID           int  `json:"step_id"`
+	BatchesPerStep   int  `json:"batches_per_step"`
```

Comment:
I like fact that we went with `num_batches` for the column in the `steps` table.  It's nice because we're basically disambiguating something that we've always conflated: the scheduling unit with the number of batches that any given step should run for.

While we are disambiguating it for the first time, can we follow through and make the name different for all instances of `Workload`/`WorkloadOperation`/`StepInfo`? Also in several places in the harness (but not in TensorPackTrial, as we discussed in the ml-ag meeting).

Reply:
Yeah sure, I was avoiding it only to avoid rebase hell in the PR coming right after this but I this it won't be _too_ bad. Additionally, in doing this upcoming PR, `batches_completed` turned out to be an extremely confusing name so I want to change it to `total_batches_processed`, too.

Reply:
Just pushed this, let me know what you think.

Reply:
looks great